### PR TITLE
Issue #26 content length error

### LINKED ISSUE #26: ServerRequest Mock: Incorrectly Setting Content Length
**Describe the bug**
When setting a `FormData` object to the body, `mocks.ServerRequest` tries to get the length (assumes it's a string) of it to set as the content type.

**To Reproduce**

```typescript
const request = Rhum.mocks.ServerRequest("/", "get", {
  body: new FormData
})
```

See error when compiled

**Expected behavior**
Should not error and (ideally) set the content length another way

**Suggested Solution(s)**
Unsure

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #26 

**Description**

* Make setting the content length account for form data as the body

    * Had to figure out how to get the content length of a form data object - took **a lot** of googling with very little code to implement D:

* Add extra tests

### CODE REVIEW

File: src/mocks/server_request.ts [hunk 1]
```diff
@@ -16,7 +16,22 @@ export const MockServerRequest = function (
       }
     }
     if (options.body) {
-      request.headers.set("Content-Length", options.body.length.toString());
+      // Used to see if the body can be stringified, or if its a formdata object
+      if (typeof options.body === "string") {
+        request.headers.set("Content-Length", options.body.length.toString());
```
Comment (crookse) [inline src/mocks/server_request.ts:21]: if it's a string do we need `toString()` ?

Comment (Guergeiro) [reply src/mocks/server_request.ts:21]: I don't think `length` is supposed to be converted to string.

Comment (Guergeiro) [reply src/mocks/server_request.ts:21]: And I think it's in octets. So a the string `hello` has length 5 but the content length should be 40.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Length

Comment (ebebbington) [reply src/mocks/server_request.ts:21]: Doesn’t .length on an array return an int? And judging by that article, I think it should be a number and not a string

Comment (Guergeiro) [reply src/mocks/server_request.ts:21]: Yup. It should be a number. If you use `.toString()` you are converting to a string.

Comment (ebebbington) [reply src/mocks/server_request.ts:21]: I was confused on what data type the content length was, will fix it 👍

**Comment by ebebbington** [comment]
Closing because some of the server request mock is going to be stripped out, making this PR redundant

**Pull request closed without merging**